proxy/proxy.go

```diff
@@ -138,9 +138,11 @@ func (p *Proxy) handle(w http.ResponseWriter, r *http.Request) error {
 		return fmt.Errorf("error forwarding request: %s", err)
 	}
 	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	source_type := http.DetectContentType(body)
 	rw := newResponseWriter(w)
 	rr := newResponseReader(resp)
-	err = p.proxyResponse(rw, rr, r.Header)
+	err = p.proxyResponse(rw, rr, r.Header, source_type)
```

Comment:
`proxyResponse` takes four arguments and this fails in CI.  Is this PR complete?

Also how does this work now that we have read `resp.Body`?  Is it OK to read the stream again?

Reply:
I think it's OK to read resp.Body again, however it will be slower than read only once. Actually I'm not familiar with golang, is there a better option to complete it?